default-fields/0000-default-fields.md

```diff
@@ -28,9 +233,29 @@ Why should we *not* do this?
 # Alternatives
 [alternatives]: #alternatives
 
-What other designs have been considered? What is the impact of not doing this?
+Just stick with builders and constructor functions.
 
 # Unresolved questions
 [unresolved]: #unresolved-questions
 
-What parts of the design are still TBD?
+## `const`
+
+Is `const` the right feature to base off?
+It means field values are just as constrained as `const` is now,
+but also prevents struct definitions from becomes bloated by complex logic that
+maybe belongs in a builder instead.
```

Comment:
So, actually, let me qualify my earlier "yes". I think it depends more on the design of the details of the feature than on the actual implementation.

If you want to investigate the details of how struct lits are compiled, you need to look at the compiler, or possibly dump the LLVM output. But really, there is nothing surprising happening atm in compilation. The only detail really FRU.

What is important, I think, is that there is no initialisation at all, it really is just a literal expression. That is, it just copies the field values into a given bit of memory.

> I think the simplest implementation would be to treat struct initialisation with field defaults as sugar for the 'real' initialiser, where defaults for missing fields are spliced in.

I agree, and using const values for default values fits well with this model - there is no extra runtime overhead at all. Anything beyond const values (I think) introduces some runtime overhead. If you allow `Default` integration or other extensions, then there is runtime overhead.

So, I agree with your original points - "complex logic that maybe belongs in a builder instead", and I'm trying to dig into the 'why' behind this point. I think expectations around runtime overhead are part of that.